### PR TITLE
feat(parser): Add S3 Object Lambda Event

### PR DESCRIPTION
**Pull request by risenberg-cyberark**
implements https://github.com/awslabs/aws-lambda-powertools-python/issues/361
also added missing imports in init file

### CODE REVIEW

**Review by heitorlessa — CHANGES_REQUESTED** [review]
Looks great! One minor typo in the docs and I can merge it.

> Note to self

For 2.0, I'd love to see how we can reuse data structure between Data classes and Parser. The only three key differences in here are:

1. Parser needs to use multi-inheritance (BaseModel, plus what's already defined)
2. Parser has some pydantic types that are more explicit so ^ could help
3. Data classes use `@property` to provide per attribute docs, so maybe there's a native way of doing this, so we could reuse code with Parser

Atm, these are the biggest raw code albeit small contributors to the package size (w/o deps)

```bash
4.0K	aws_lambda_powertools/utilities/__init__.py
348K	aws_lambda_powertools/utilities/data_classes
192K	aws_lambda_powertools/utilities/parser
108K	aws_lambda_powertools/utilities/parameters
 96K	aws_lambda_powertools/utilities/idempotency
 48K	aws_lambda_powertools/utilities/validation
 44K	aws_lambda_powertools/utilities/batch
 40K	aws_lambda_powertools/utilities/typing
```

File: docs/utilities/parser.md [hunk 1]
```diff
@@ -149,17 +149,18 @@ Use this standalone function when you want more control over the data validation
...
-**SnsModel** |  Lambda Event Source payload for Amazon Simple Notification Service
+| Model name                 | Description                                                        |
+| -------------------------- | ------------------------------------------------------------------ |
+| **DynamoDBStreamModel**    | Lambda Event Source payload for Amazon DynamoDB Streams            |
+| **EventBridgeModel**       | Lambda Event Source payload for Amazon EventBridge                 |
+| **SqsModel**               | Lambda Event Source payload for Amazon SQS                         |
+| **AlbModel**               | Lambda Event Source payload for Amazon Application Load Balancer   |
+| **CloudwatchLogsModel**    | Lambda Event Source payload for Amazon CloudWatch Logs             |
+| **S3Model**                | Lambda Event Source payload for Amazon S3                          |
+| **S3ObjectLambdaEvent**    | Lambda Event Source payload for Amazon S3 Object Access            |
```
[the remark this answers is not in the document]
Comment (risenberg-cyberark) [reply docs/utilities/parser.md]: thx! pushed the code fix

[the remark this answers is not in the document]
Comment (risenberg-cyberark) [reply docs/utilities/parser.md]: @heitorlessa BTW, more key differences between parser and validator are:
1. We can use Literal types - i thought of using them in this PR but the documentation wasnt clear 100%.
2. We can add validators and root validator. in this case for instance, the userName exists only in some cases - i could add that check to the schema.

**Comment by heitorlessa** [comment]
Oh I know about the added value of Parser, and Literal types are part of
point 1 ;)

That note to myself is to remember what a base class for both could look
like.

**Review by heitorlessa — APPROVED** [review]
Thank you once again! Merging now

**Pull request merged**